src/components/textbox/textbox.test.tsx

```diff
@@ -0,0 +1,607 @@
+import StyledInput from "../../__internal__/input/input.style";
+
+jest.mock("../../__internal__/utils/logger");
+
+const mockedGuid = "mocked-guid";
+jest.mock("../../__internal__/utils/helpers/guid");
+
+(createGuid as jest.MockedFunction<typeof createGuid>).mockReturnValue(
+  mockedGuid
+);
+
+let loggerSpy: jest.SpyInstance<void, [message: string]> | jest.Mock;
+
+beforeEach(() => {
+  loggerSpy = jest.spyOn(Logger, "deprecate");
+});
+
+afterEach(() => {
+  loggerSpy.mockRestore();
+});
+
+afterAll(() => {
+  loggerSpy.mockClear();
+});
+
+test("should display deprecation warning once when rendered as uncontrolled", () => {
+  render(<Textbox name="my-textbox" defaultValue="test" />);
+
+  expect(loggerSpy).toHaveBeenCalledWith(
+    "Uncontrolled behaviour in `Textbox` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+
+  expect(loggerSpy).toHaveBeenCalledTimes(1);
+});
+
+testStyledSystemMargin(
+  (props) => <Textbox {...props} />,
+  undefined,
+  (component) => component.find(FormFieldStyle),
+  { modifier: "&&&" }
+);
+
+describe(`when the characterLimit prop is passed`, () => {
+  test.each([2, 3, 4])("renders a character counter", (characterLimit) => {
+    const valueString = "foo";
+    const limitMinusValue = characterLimit - valueString.length >= 0;
+    render(<Textbox value={valueString} characterLimit={characterLimit} />);
+    const underCharacters =
+      characterLimit - valueString.length === 1 ? "character" : "characters";
+    const overCharacters =
+      valueString.length - characterLimit === 1 ? "character" : "characters";
+
+    expect(
+      screen.getByText(
+        `${
+          limitMinusValue
+            ? `${characterLimit - valueString.length} ${underCharacters} left`
+            : `${
+                valueString.length - characterLimit
+              } ${overCharacters} too many`
+        }`,
+        { selector: '[aria-hidden="true"]' }
+      )
+    ).toBeInTheDocument();
+  });
+
+  test("visually hidden hint should have id generated via guid", () => {
+    render(<Textbox value="foo" characterLimit={73} />);
+    expect(
+      screen.getByText("You can enter up to 73 characters", {
+        selector: '[data-element="visually-hidden-hint"]',
+      })
+    ).toHaveAttribute("id", mockedGuid);
+  });
+
+  test("that visually hidden hint id should be referenced in input's aria-described by", () => {
+    render(<Textbox value="foo" characterLimit={73} />);
+    expect(screen.getByRole("textbox")).toHaveAttribute(
+      "aria-describedby",
+      mockedGuid
+    );
+  });
+
+  test("renders a counter with an over limit warning", () => {
+    render(<Textbox value="test string" characterLimit={10} />);
+
+    expect(
+      screen.getByText("1 character too many", {
+        selector: '[aria-hidden="true"]',
+      })
+    ).toHaveStyleRule("color", "var(--colorsSemanticNegative500)");
+  });
+});
+
+describe("refs", () => {
```

Comment:
suggestion(non-blocking): this describe block isn't adding much value

We should probably either be more verbose with the description or drop the describe block

Reply:
fair comment on this, and the other similar ones - I've already removed/collapsed a lot of `describe`s and left what I thought was a sensible minimum, but happy to look again and get rid of more of them.

Reply:
Agree massive improvement on what we had before but I think we can flatten these one's further and just use `test` here